Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -1347,8 +1351,16 @@ extension Tensor {
 
   /// Returns `true` if all scalars of the given 1-D tensor are in the range `[-rank, rank)`.
   @usableFromInline
-  internal func areAxesInRange(_ axes: Tensor<Int32>) -> Bool {
-    precondition(axes.rank == 1, "Axes must have rank 1.")
+  internal func areAxesInRange(
+    _ axes: Tensor<Int32>,
+    file: StaticString = #file,
+    line: UInt = #line
+  ) -> Bool {
+    precondition(
```

Comment:
I thought you were going to stick to improving error messages with this commit?

The comment still implies the old precondition was correct; you've expanding the method to mean that it works on "scalar tensors," so you need to update the comment to reflect that.  If you do that, consider whether the convenience offered by the expansion is enough to outweigh the complexity added by complicating the semantics.